tests/probes/symlink/test_offline_mode_symlink.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+# Copyright 2015 Red Hat Inc., Durham, North Carolina.
+# All Rights Reserved.
+#
+# OpenScap Probes Test Suite.
+
+. $builddir/tests/test_common.sh
+
+set -e -o pipefail
+
+function test_offline_mode_symlink {
+    probecheck "symlink" || return 255
+
+    DF="test_offline_mode_symlink.xml"
+    RF="results.xml"
+
+    rm -f $RF
+
+    tmpdir=$(mktemp -t -d "test_symlink.XXXXXX")
+    touch $tmpdir/some_file
+    touch $tmpdir/file_to_remove
+    ln -s /some_file $tmpdir/normal_symlink
+    ln -s some_file $tmpdir/normal_symlink_relative
+    ln -s /file_to_remove $tmpdir/broken_symlink
+    ln -s file_to_remove $tmpdir/broken_symlink_relative
+    rm -f $tmpdir/file_to_remove
+    ln -s /circular_symlink $tmpdir/circular_symlink
+    touch $tmpdir/other_file
+    ln -s /other_file $tmpdir/middle_symlink
+    ln -s /middle_symlink $tmpdir/chained_symlink
+    mkdir $tmpdir/etc
+    touch $tmpdir/etc/shadow
+    ln -s /etc/shadow $tmpdir/etc/passwd
+    touch $tmpdir/file_in_root
+    ln -s ../../../../../file_in_root $tmpdir/path_traversal_symlink
+    ln -s ./some_file $tmpdir/symlink_with_period_in_path
+    touch /tmp/symlinktarget
+    ln -s /tmp/symlinktarget /tmp/symlinktest
```

Comment:
These two cause problems if they exist from the previous test run - and if a test run fails, those links won't be deleted. I recommend using `mktemp` without the `-d` here.

Reply:
I would leave this for future improvements. In current state it is not possible to use mktemp here, because the file name is hardcoded in the generated oval. And if the file exists from the previous run, that actually doees not matter. The probe should not be able to see files if working correctly.

Reply:
We actually need to address this, as there is `set -e` effective, os if there is already `/tmp/symlinktarget`, the test will abort on this line. I wouldn't worry about hardcoding either - you can parametrize those names in the same manner as `tmpdir` is parametrized.

Reply:
In your last commit, you have moved the cleanup commands. However, you need to make sure that `/tmp/symlinktest` and `/tmp/symlinktarget` are not there *before* you try to create them. And I am pretty sure that if you remove results before actually checking them, the test will fail.